stan/math/prim/mat/fun/gp_matern52_cov.hpp

```diff
@@ -0,0 +1,323 @@
+#ifndef STAN_MATH_PRIM_MAT_FUN_GP_MATERN52_COV_HPP
+#define STAN_MATH_PRIM_MAT_FUN_GP_MATERN52_COV_HPP
+
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <stan/math/prim/mat/fun/sqrt.hpp>
+#include <stan/math/prim/scal/err/check_finite.hpp>
+#include <stan/math/prim/scal/err/check_nonnegative.hpp>
+#include <stan/math/prim/scal/err/check_not_nan.hpp>
+#include <stan/math/prim/scal/err/check_positive_finite.hpp>
+#include <stan/math/prim/scal/fun/square.hpp>
+#include <stan/math/prim/scal/fun/squared_distance.hpp>
+#include <stan/math/prim/scal/meta/return_type.hpp>
+#include <stan/math/prim/scal/fun/divide.hpp>
+#include <cmath>
+#include <vector>
+
+namespace stan {
+namespace math {
+
+/**
+ * Returns a Matern 5/2 covariance matrix with one input vector
+ *
+ * \f[ k(x, x') = \sigma^2\bigg(1 +
+ *  \frac{\sqrt{5}d(x, x')}{l} + \frac{5d(x, x')^2}{3l^2}\bigg)
+ *   exp\bigg(-\frac{5 d(x, x')}{l}\bigg)
+ * \f]
+ *
+ * where \f$ d(x, x') \f$ is the Euclidean distance.
+ *
+ * @tparam T_x type of elements contained in vector x
+ * @tparam type of element of sigma, marginal standard deviation
```

Comment:
This comment is wrong. This should have been caught by doxygen, but `type` should be `T_s`.